drivers/espi/espi_it8xxx2.c

```diff
@@ -0,0 +1,1278 @@
+/*
+ * Copyright (c) 2021 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_espi
+
+#include <assert.h>
+#include <drivers/espi.h>
+#include <drivers/gpio.h>
+#include <kernel.h>
+#include <soc.h>
+#include "soc_espi.h"
+#include "espi_utils.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(espi, CONFIG_ESPI_LOG_LEVEL);
+
+#define ESPI_IT8XXX2_GET_GCTRL_BASE \
+	((struct gctrl_it8xxx2_regs *)DT_REG_ADDR(DT_NODELABEL(gctrl)))
+
+#define IT8XXX2_ESPI_IRQ     DT_INST_IRQ_BY_IDX(0, 0, irq)
+#define IT8XXX2_ESPI_VW_IRQ  DT_INST_IRQ_BY_IDX(0, 1, irq)
+#define IT8XXX2_KBC_IBF_IRQ  DT_INST_IRQ_BY_IDX(0, 2, irq)
+#define IT8XXX2_KBC_OBE_IRQ  DT_INST_IRQ_BY_IDX(0, 3, irq)
+#define IT8XXX2_PMC1_IBF_IRQ DT_INST_IRQ_BY_IDX(0, 4, irq)
+#define IT8XXX2_PORT_80_IRQ  DT_INST_IRQ_BY_IDX(0, 5, irq)
+
+/* General Capabilities and Configuration 1 */
+#define IT8XXX2_ESPI_MAX_FREQ_MASK GENMASK(2, 0)
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_20 0
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_25 1
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_33 2
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_50 3
+#define IT8XXX2_ESPI_CAPCFG1_MAX_FREQ_66 4
+
+#define IT8XXX2_ESPI_PC_READY_MASK  BIT(1)
+#define IT8XXX2_ESPI_VW_READY_MASK  BIT(1)
+#define IT8XXX2_ESPI_OOB_READY_MASK BIT(1)
+#define IT8XXX2_ESPI_FC_READY_MASK  BIT(1)
+
+#define IT8XXX2_ESPI_INTERRUPT_ENABLE          BIT(7)
+#define IT8XXX2_ESPI_TO_WUC_ENABLE             BIT(4)
+#define IT8XXX2_ESPI_VW_INTERRUPT_ENABLE       BIT(7)
+#define IT8XXX2_ESPI_INTERRUPT_PUT_PC          BIT(7)
+
+#define IT8XXX2_ESPI_UPSTREAM_INTERRUPT_ENABLE BIT(5)
+#define IT8XXX2_ESPI_UPSTREAM_CHANNEL_DISABLE  BIT(2)
+#define IT8XXX2_ESPI_UPSTREAM_DONE             BIT(1)
+
+#define IT8XXX2_ESPI_PUT_OOB_STATUS            BIT(7)
+#define IT8XXX2_ESPI_PUT_OOB_INTERRUPT_ENABLE  BIT(7)
+
+#define IT8XXX2_ESPI_INPUT_PAD_GATING          BIT(6)
+
+struct espi_it8xxx2_config {
+	uintptr_t base_espi_slave;
+	uintptr_t base_espi_vw;
+	uintptr_t base_espi_queue1;
+	uintptr_t base_espi_queue0;
+	uintptr_t base_ec2i;
+	uintptr_t base_kbc;
+	uintptr_t base_pmc;
+};
+
+struct espi_it8xxx2_data {
+	sys_slist_t callbacks;
+};
+
+/* Driver convenience defines */
+#define DRV_CONFIG(dev) ((const struct espi_it8xxx2_config *)(dev)->config)
+#define DRV_DATA(dev) ((struct espi_it8xxx2_data *)(dev)->data)
+
+struct vw_channel_t {
+	uint8_t  vw_index;      /* VW index of signal */
+	uint8_t  level_mask;    /* level bit of signal */
+	uint8_t  valid_mask;    /* valid bit of signal */
+};
+
+struct vwidx_isr_t {
+	void (*vwidx_isr)(const struct device *dev, uint8_t update_flag);
+	uint8_t vw_index;
+};
+
+struct espi_isr_t {
+	void (*espi_isr)(const struct device *dev, bool enable);
+	bool asserted_flag;
```

Comment:
I suggest an enum here to make the code clearer.  Something like
```
enum isr_type {
    DEASSERTED_FLAG = 0,
    ASSERTED_FLAG = 1,
}
static const struct espi_isr_t espi_isr_list[] = {
	[0] = {espi_it8xxx2_peripheral_ch_en_isr, ASSERTED_FLAG},
	[4] = {espi_it8xxx2_peripheral_ch_en_isr, DEASSERTED_FLAG},
```

Reply:
Done.

Reply:
Done.